eth_account/ledger.py

```diff
@@ -0,0 +1,193 @@
+    AttributeDict,
+)
+
+from eth_account.internal.transactions import (
+    encode_transaction,
+    serializable_unsigned_transaction_from_dict,
+)
+
+from eth_utils.curried import (
+    keccak,
+)
+
+ETH_DERIVATION_PATH_PREFIX = "m/44'/60'/0'/"
+
+
+class LedgerAccount:
+    '''
+    Ledger Ethereum App Protocol Spec is located at:
+    <https://github.com/LedgerHQ/blue-app-eth/blob/master/doc/ethapp.asc>
+
+    References:
+    - https://github.com/LedgerHQ/blue-app-eth/blob/master/getPublicKey.py
+    - https://github.com/bargst/pyethoff/blob/master/tx_sign.py
+
+    '''
+
+    def __init__(self, account_id=0, address='', path_prefix=ETH_DERIVATION_PATH_PREFIX):
+        self.device = None
+        self.path_prefix = path_prefix
+
+        if eth_utils.is_address(address):
+            self.account_id = self.get_account_id(address)
+            self.address = address
+        else:
+            self.account_id = account_id
+            self.address = self.get_address(account_id)
+
+
+    def _path_to_bytes(self, path):
+        '''
+        This function convert a bip32 string path to a bytes format used by the
+        Ledger device. Path are of the form :
+
+            m/ ? / ? / ? / ? .... with an arbitrary depth
+
+        In most case, bip44 is used as a subset of bip32 with path of the form:
+
+        m / purpose' / coin_type' / account' / change / address_index
+
+        Apostrophe in the path indicates that bip32 hardened derivation is used.
+
+        Ledger expect the following bytes input:
+        1. Number of BIP 32 derivations to perform (max 10)
+        2. First derivation index (big endian)
+        3. Second derivation index (big endian)
+        ...
+        Nth. Last derivation index (big endian)
+
+        References:
+        - https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
+        - https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
+        '''
+        assert(path.startswith('m/'))
+
+        elements = path.split('/')[1:]
+        depth = len(elements)
+
+        result = bytes([depth]) # Number of BIP 32 derivations to perform (max 10)
+
+        for derivation_index in elements:
+            # For each derivation index in the path check if it is hardened
+            hardened = "'" in derivation_index
+            index = int(derivation_index.strip("'"))
+
+            if hardened:
+                # See bip32 spec for hardened derivation spec
+                index = 0x80000000 | index
+
+            # Append index to result as a big-endian (>) unsigned int (I)
+            result += struct.pack('>I', index)
+
+        return result
+
+    def _send_to_device(self, apdu):
+        try:
+            if self.device is None:
+                self.device = ledgerblue.comm.getDongle(debug=False)
+
+            reply = self.device.exchange(apdu, timeout=60)
+        except:
+            raise Exception('Something went wrong when communicating with device')
```

Comment:
Im not too keen on masking the underlying exception.

Reply:
Ok, I will remove the try catch